main.go

```diff
@@ -55,11 +56,13 @@ func main() {
 	// If not enabled multithreading, server will run on a single core.
 	var numCores int
 	if config.EnableMultiThreading {
+		serverErrCh = make(chan error, 1)
+		logr.Debug("The DiceDB server has started in multi-threaded mode.", slog.Int("number of cores", numCores))
 		numCores = runtime.NumCPU()
```

Comment:
why did we decided to move this log above `numCores = runtime.NumCPU()`? it'll always print number of cores as 0